src/controllers/setup_test.go

```diff
@@ -66,6 +68,20 @@ func createTestSecret(secretName string, namespace string, labels map[string]str
 	}
 }
 
+func createTestServiceAccount(name string, namespace string, labels map[string]string) {
+	serviceAccount := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    labels,
+		},
+	}
+	_, err := fakeClient.CoreV1().ServiceAccounts(namespace).Create(context.TODO(), serviceAccount, metav1.CreateOptions{})
+	if err != nil {
+		panic(err)
+	}
+}
```

Comment:
Maybe move this to the `mocks` package? Like this:

https://github.com/dana-team/platform-backend/blob/42cd557e3785424ba6c893f8274ea22a092f2ad8/src/utils/testutils/mocks/adapter.go#L86-L93